cfn-resources/cluster/cmd/resource/mappings.go

```diff
@@ -646,3 +646,51 @@ func setClusterRequest(currentModel *Model) (*admin.AdvancedClusterDescription,
 	clusterRequest.TerminationProtectionEnabled = currentModel.TerminationProtectionEnabled
 	return clusterRequest, nil
 }
+
+func AddReplicationSpecIds(src, dest []admin.ReplicationSpec) *[]admin.ReplicationSpec {
+	zoneToID := map[string]string{}
+	providerRegionToID := map[string]string{}
+	usedIds := map[string]bool{}
+
+	for _, spec := range src {
+		specID := spec.GetId()
+		if specID == "" {
+			continue
+		}
+		if zoneName := spec.GetZoneName(); zoneName != "" {
+			zoneToID[zoneName] = specID
+		}
+		if providerRegion := asProviderRegion(spec); providerRegion != "" {
+			providerRegionToID[providerRegion] = spec.GetId()
+		}
+	}
+	for i, spec := range dest {
+		specID := spec.GetId()
+		if specID != "" {
+			continue
+		}
+		idZone, foundZone := zoneToID[spec.GetZoneName()]
+		zoneUsed := usedIds[idZone]
+		if foundZone && !zoneUsed {
+			usedIds[idZone] = true
+			dest[i].SetId(idZone)
+			continue
+		}
+		idProvider, foundProvider := providerRegionToID[asProviderRegion(spec)]
+		providerUsed := usedIds[idProvider]
+		if foundProvider && !providerUsed {
+			usedIds[idProvider] = true
+			dest[i].SetId(idProvider)
+			continue
+		}
+	}
+	return &dest
+}
+
+func asProviderRegion(spec admin.ReplicationSpec) string {
+	configs := spec.GetRegionConfigs()
+	if len(configs) == 0 {
+		return ""
+	}
+	return fmt.Sprintf("%s-%s", configs[0].GetProviderName(), configs[0].GetRegionName())
+}
```

Comment:
more as an FYI: there can be replication specs with multiple region configs (case of [multi cloud](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster#example-multi-cloud-cluster)). 
My understanding is that if zone name is not defined the cluster is either a replica set or SHARDED with only a single replication spec, so this logic might be redundant.

Reply:
Thanks @AgustinBettati 
Yes, I realized it was a list, but I wasn't sure about the guaranteed order or behavior when adding more region configs.
Therefore, I implemented a "naive" solution only using the 1st region config in the list. 

WDYT? Would it make sense to match on multiple region configs? Is there any guarantee on ordering?

Reply:
I don't think API guarantees order so it's better to not rely on that

Reply:
I don't think the API guarantees order, it might be better to not rely on that

Reply:
An alternative would be to order by the Provider & RegionName and concat these together. 
However, I think keeping it as is works better:

1. I believe there can be a case where the user **adds** an additional regionConfig, if we match on all region configs then the replication ID will never match since the "new" regionConfig will not exist in the currentCluster
2. Keeps the matching logic simpler. 

Happy to hear suggestions: @maastha @AgustinBettati 😊

Reply:
Ok. Choosing to merge as is. Thanks for all the feedback ❤️